enumerable.rb

```diff
@@ -0,0 +1,90 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    i = 0
+    while i < check_self.length
+      yield check_self[i]
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    return to_enum :my_each_with_index unless block_given?
+
+    i = 0
+    my_each do |element|
+      yield element, i
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    return enum_for :my_select unless block_given?
+
+    Enumerator.new do |y|
+      my_each { |item| y << item if yield item }
+    end.to_a
+  end
+
+  def my_all?(initial = nil)
+    result = true
+    if block_given?
+      my_each { |element| result &= (yield element) }
+    elsif initial
+      my_each { |element| result &= pattern == element }
+    else
+      my_each { |element| result &= element }
+    end
+    result
+  end
+
+  def my_any?
+    my_each { |item| return true if block_given? ? yield(item) : item }
+    false
+  end
+
+  def my_none?
+    my_each do |item|
+      return false if block_given? ? yield(item) : item
+    end
+    true
+  end
+
+  def my_count(item = nil)
+    count = 0
+    if block_given?
+      my_each { |i| count += 1 if yield(i) == true }

```

Comment:
```
ary = [1, 2, 4, 2]
ary.count               #=> 4
ary.count(2)            #=> 2
ary.count{ |x| x%2==0 } #=> 3
```

- [ ] my_count returns the number of items in enum through enumeration
- [ ] my_count counts the number of items in enum that are equal to item if an argument is given
- [ ] my_count counts the number of elements yielding a true value if a block is given